rpc/src/v1/impls/snapshot.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018-2019 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::fs;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use ccore::{BlockChainClient, BlockId, SnapshotClient as CoreSnapshotClient};
+use ctypes::BlockHash;
+use primitives::H256;
+
+use jsonrpc_core::Result;
+
+use super::super::errors;
+use super::super::traits::Snapshot;
+use super::super::types::BlockNumberAndHash;
+
+pub struct SnapshotClient<C>
+where
+    C: BlockChainClient + CoreSnapshotClient, {
+    client: Arc<C>,
+}
+
+impl<C> SnapshotClient<C>
+where
+    C: BlockChainClient + CoreSnapshotClient,
+{
+    pub fn new(client: Arc<C>) -> Self {
+        SnapshotClient {
+            client,
+        }
+    }
+}
+
+impl<C> Snapshot for SnapshotClient<C>
+where
+    C: BlockChainClient + CoreSnapshotClient + 'static,
+{
+    fn get_snapshot_list(&self) -> Result<Vec<BlockNumberAndHash>> {
+        let ids = {
+            if let Some(snapshot_dir) = self.client.snapshot_dir() {
```

Comment:
Change this to early exit pattern.
```rust
let snapshot_dir = if let Some(snapshot_dir) = self.client.snapshot_dir() { 
  snapshot_dir 
} else { 
  return Ok(Vec::new()) 
};
```

Reply:
I think an early return is not a good choice here, but I think I can improve this. Let me fix it.